<launch>
    <arg name="data_file"/>
    <arg name="pub_frequency" default="30"/>
    <arg name="data_frequency" default="200"/>
    <arg name="cable_topic" default="/tensegrity/cables"/>
    <arg name="endcap_noise_sigma" default="0.0"/>
    <arg name="endcap_miss_probability" default="0.0"/>
    <arg name="cables_noise_sigma" default="0.0"/>
    <arg name="visualize" default="true"/>
    <arg name="endcaps_topic" default="/tensegrity/endcaps/proposed"/>
    <arg name="output_dir" default="$(env TENSEGRITY_ROS)/data/test/"/>
    <arg name="filename_prefix" default="test"/>
    <arg name="Nbars_ratio" default="1.0"/>
    <arg name="cable_map_filename" default="cable_map.txt"/>
    <arg name="use_cable_sensors" default="true"/>
    <arg name="autostart" default="false"/>


    <include file="$(find interface)/launch/simulated_data.launch">
        <arg name="data_file" default="$(arg data_file)"/>
        <arg name="pub_frequency" default="$(arg pub_frequency)"/>
        <arg name="data_frequency" default="200"/>
        <arg name="loop" default="false"/>
        <arg name="endcaps_topic" default="$(arg endcaps_topic)"/>
        <arg name="red_endcaps_topic" default="/tensegrity/endcap/red/positions"/>
        <arg name="green_endcaps_topic" default="/tensegrity/endcap/green/positions"/>
        <arg name="blue_endcaps_topic" default="/tensegrity/endcap/blue/positions"/>
        <arg name="tensegrity_gt_topic" default="/tensegrity/gt"/>
        <arg name="cable_topic" default="$(arg cable_topic)"/>
        <arg name="endcap_noise_sigma" default="$(arg endcap_noise_sigma)"/>
        <arg name="endcap_miss_probability" default="$(arg endcap_miss_probability)"/>
        <arg name="cables_noise_mu" default="0.0"/>
        <arg name="cables_noise_sigma" default="$(arg cables_noise_sigma)"/>
        <arg name="data_scale_factor" default="0.10"/>
        <arg name="sim_clock" default="false"/>
        <arg name="endcap_swap_probability" default="0.0"/>
        <arg name="visualize" default="$(arg visualize)"/>
        <arg name="filename_prefix" default="$(arg output_dir)/$(arg filename_prefix)"/>
    </include>

    <node pkg="estimation" type="chev_endcaps" name="chev_endcaps" required="true" output="screen">
        <param name="node_id" value="/nodes/cheb_endcaps"/>
        <param name="tensegrity_bars_topicname" value="$(arg endcaps_topic)"/>
        <param name="publisher_node_id" value="/node/simulation"/>
        <param name="cheb_json_filename" value="$(arg output_dir)/$(arg filename_prefix)_cheb_endcaps_bars.json"/>
        <param name="cable_unit_conversion" value="1"/>
        <param name="cable_map_filename" value="$(arg cable_map_filename)"/>
        <param name="sensors_topicname" value="/tensegrity/sensors"/>
        <param name="other_frame" value="world"/>
        <param name="world_frame" value="world"/>
        <param name="use_cable_sensors" value="$(arg use_cable_sensors)"/>
        <param name="Nbars_ratio" value="$(arg Nbars_ratio)"/>
    </node>

    <node if="$(arg autostart)" pkg="interface" type="status_wait_and_run.py" name="status_wait_and_run" output="screen">
        <param name="nodes_to_wait" type="yaml" 
            value="[
              ]"
            />
        <param name="nodes_to_wait_ready" type="yaml" 
            value="[
                /nodes/cheb_endcaps/status/current,
                /node/simulation/status/current
            ]"
            />
        <param name="nodes_to_run" type="yaml" 
            value="[
                /node/simulation/status/change
            ]"
            />
  </node>
</launch>